istioctl/cmd/internal-debug.go

```diff
@@ -139,57 +139,74 @@ By default it will use the default serviceAccount from (istio-system) namespace
 				TypeUrl: TypeDebug,
 			}
 
-			if centralOpts.Xds == "" {
-				svc, err := kubeClient.CoreV1().Services(istioNamespace).Get(context.Background(), istiodServiceName, metav1.GetOptions{})
-				if err != nil {
-					return fmt.Errorf("please specify %q as %v", "--xds-address", err)
+			svc, err := kubeClient.CoreV1().Services(istioNamespace).Get(context.Background(), istiodServiceName, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("please specify %q as %v", "--xds-address", err)
+			}
+			istiodNS, selector, serr := polymorphichelpers.SelectorsForObject(svc)
+			if serr != nil {
+				return fmt.Errorf("please specify %q as we cannot attach to %T: %v", "--xds-address", svc, serr)
+			}
+			options := metav1.ListOptions{LabelSelector: selector.String()}
+			podList, plerr := kubeClient.CoreV1().Pods(istiodNS).List(context.TODO(), options)
```

Comment:
You can use `kubeClient.GetIstioPods` directly

Reply:
@irisdingbj Okay, thanks Iris, I will check it!